iothub/service/src/IotHubServiceClient.cs

```diff
@@ -49,20 +49,28 @@ protected IotHubServiceClient()
         public IotHubServiceClient(string connectionString, IotHubServiceClientOptions options = default)
         {
             Argument.AssertNotNullOrWhiteSpace(connectionString, nameof(connectionString));
-
-            if (options == null)
-            {
-                options = new IotHubServiceClientOptions();
-            }
+            IotHubServiceClientOptions clientOptions = options != null
```

Comment:
Consider making this a private function since this clone logic is written multiple times in this file

Reply:
Maybe even have an internal method ```clone``` on each options class that returns this copy